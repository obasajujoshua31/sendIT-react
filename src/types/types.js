export const signUpUser = "SIGN_UP_USER";
export const signInUser = "SIGN_IN_USER";
export const createOrder = "CREATE_ORDER";
export const createOrderSuccess = "CREATE_ORDER_SUCCESS";
export const createOrderFailure = "CREATE_ORDER_FAILURE";
export const viewOrders = "VIEW_ORDERS";
export const viewOrder = "VIEW_ORDER";
export const updateParcel = "UPDATE_PARCELS";
export const updateLocation = "UPDATE_PARCEL_LOCATION";
export const cancelOrder = "CANCEL_ORDER";
export const updateParcelStatus = "UPDATE_PARCEL_STATUS";
export const signInSuccess = "SIGN_IN_SUCCESS";
export const signInFailure = "SIGN_IN_FAILURE";
export const signUpSuccess = "SIGN_UP_SUCCESS";
export const signUpFailure = "SIGN_UP_FAILURE";
export const isLoggedOut = "LOGGED_OUT";
export const isAdmin = "IS_ADMIN";
