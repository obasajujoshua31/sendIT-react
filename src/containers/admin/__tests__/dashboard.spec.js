import React from "react";
import {
  AdminDashboard,
  mapDispatchToProps,
  mapStateToProps
} from "../dashboard";
import { shallow } from "enzyme";
import sinon from "sinon";
import { signInSuccess, signInFailure } from "../../../types/types";

const props = {
  userParcels: [],
  loadParcels: jest.fn(),
  history: {
    push: jest.fn()
  },
  authStatus: signInSuccess
};

const mockParcels = {
  data: [
    {
      parcel: {
        status: "PLACED"
      }
    },
    {
      parcel: {
        status: "DELIVERED"
      }
    },
    {
      parcel: {
        status: "TRANSITING"
      }
    },
    {
      parcel: {
        status: "CANCELLED"
      }
    }
  ]
};

const mockStore = {
  users: {
    singInStatus: signInSuccess
  },
  parcels: {
    parcels: ["Beans"]
  }
};
const wrapper = shallow(<AdminDashboard {...props} />);

describe("Test for User AdminDashboard", () => {
  it("should render AdminDashboard component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should dispatch loadparcels", () => {
    const dispatchSpy = sinon.spy();
    const dispatchResult = mapDispatchToProps(dispatchSpy);
    dispatchResult.loadParcels();
    expect(dispatchSpy.callCount).toEqual(1);
  });

  it("should return new object for mapstate to props", () => {
    const stateToPropsResult = mapStateToProps(mockStore);
    expect(stateToPropsResult.userParcels).toEqual(["Beans"]);
    expect(stateToPropsResult.authStatus).toEqual(signInSuccess);
  });

  it("should ", () => {
    const dashboardWrapper = shallow(<AdminDashboard {...props} />);
    dashboardWrapper.setProps({
      userParcels: mockParcels
    });
    expect(dashboardWrapper).toMatchSnapshot();
  });

  it("should call history.push", () => {
    const dashboardWrapper = shallow(<AdminDashboard {...props} />);
    dashboardWrapper.setState({
      authStatus: ""
    });
    dashboardWrapper.setProps({
      authStatus: signInFailure
    });
    expect(dashboardWrapper.instance().props.history.push).toBeCalled();
  });

  it("render when there is no change in props", () => {
    const dashboardWrapper = shallow(<AdminDashboard {...props} />);
    dashboardWrapper.setState({
      authStatus: signInFailure
    });
    dashboardWrapper.setProps({
      authStatus: signInFailure
    });
    expect(dashboardWrapper).toMatchSnapshot();
  });
});
