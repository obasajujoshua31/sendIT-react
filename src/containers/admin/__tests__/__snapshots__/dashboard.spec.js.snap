// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for User AdminDashboard render when there is no change in props 1`] = `
<Fragment>
  <div
    className="account-wrapper"
  >
    <div
      className="box-group"
    >
      <div
        className="container"
      >
        <div
          className="boxes"
        >
          <div
            className="box"
          >
            <span
              className="content"
            >
              Delivered Orders
            </span>
            <span
              className="content-summary"
            >
              0
            </span>
          </div>
          <div
            className="box"
          >
            <span
              className="content"
            >
              Placed Orders
            </span>
            <span
              className="content-summary"
            >
              0
            </span>
          </div>
        </div>
        <div
          className="boxes"
        >
          <div
            className="box"
          >
            <span
              className="content"
            >
              Cancelled Orders
            </span>
            <span
              className="content-summary"
            >
              0
            </span>
          </div>
          <div
            className="box"
          >
            <span
              className="content"
            >
              Transiting Orders
            </span>
            <span
              className="content-summary"
            >
              0
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</Fragment>
`;

exports[`Test for User AdminDashboard should  1`] = `
<Fragment>
  <div
    className="account-wrapper"
  >
    <div
      className="box-group"
    >
      <div
        className="container"
      >
        <div
          className="boxes"
        >
          <div
            className="box"
          >
            <span
              className="content"
            >
              Delivered Orders
            </span>
            <span
              className="content-summary"
            >
              0
            </span>
          </div>
          <div
            className="box"
          >
            <span
              className="content"
            >
              Placed Orders
            </span>
            <span
              className="content-summary"
            >
              0
            </span>
          </div>
        </div>
        <div
          className="boxes"
        >
          <div
            className="box"
          >
            <span
              className="content"
            >
              Cancelled Orders
            </span>
            <span
              className="content-summary"
            >
              0
            </span>
          </div>
          <div
            className="box"
          >
            <span
              className="content"
            >
              Transiting Orders
            </span>
            <span
              className="content-summary"
            >
              0
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</Fragment>
`;

exports[`Test for User AdminDashboard should render AdminDashboard component 1`] = `
<div
  className="container"
>
  <div
    className="table-container"
  >
    <div
      className="container-not-found"
    >
      <MDSpinner
        color1="rgb(66, 165, 245)"
        color2="rgb(239, 83, 80)"
        color3="rgb(253, 216, 53)"
        color4="rgb(76, 175, 80)"
        duration={1333}
        size={28}
      />
    </div>
  </div>
</div>
`;
